tokio/tests/task_blocking.rs

```diff
@@ -176,3 +176,54 @@ fn can_shutdown_now_in_runtime() {
         rt.shutdown_background();
     });
 }
+
+#[test]
+fn coop_disabled_in_block_in_place() {
+    let mut outer = tokio::runtime::Builder::new()
+        .threaded_scheduler()
+        .build()
+        .unwrap();
+
+    let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
+    for i in 0..200 {
+        tx.send(i).unwrap();
+    }
+    drop(tx);
+
+    outer
+        .block_on(async move {
+            tokio::spawn(async move {
+                tokio::task::block_in_place(move || {
+                    futures::executor::block_on(async move {
+                        use tokio::stream::StreamExt;
+                        assert_eq!(rx.fold(0, |n, _| n + 1).await, 200);
+                    })
+                })
+            })
+            .await
+        })
+        .unwrap();
+}
+
+#[test]
+fn coop_disabled_in_block_in_place_in_block_on() {
```

Comment:
This test would hang indefinitely with the old code.

Reply:
Elsewhere, we have some tests that run code that might hang in a separate thread, and assert it joins after a minute or so. Does it make sense to do that here?

Reply:
Good idea! Take a look at 7ef5c72d16976cbc178dc2ef6587714c314b4483?